doc/experimental/phase1/oauth2-proxy.md

```diff
@@ -0,0 +1,1350 @@
+# Migration from Verrazzano Auth proxy to OAuth2 Proxy
+
+### Version: v0.0.1-draft
+This document explains how to migrate from the Verrazzano Auth proxy to the [OAuth2 Proxy](https://github.com/oauth2-proxy/oauth2-proxy).
+
+## Summary of steps
+1. Prepare for installation of OAuth2 Proxy.
+2. Install OAuth2 Proxy.
+3. Shutdown Verrazzano auth proxy and Verrazzano monitoring operator.
+4. NGINX ingress controller configuration changes.
+5. Migrate each console to use OAuth2 Proxy.
+6. Update Fluentd config.
+7. Remove Verrazzano auth proxy from the cluster.
+
+## 1. Prepare for installation of OAuth2 Proxy 
+Before installing the OAuth2 Proxy, you must do the following:
+
+1. Create a Keycloak TLS secret.
+2. Create a OAuth2 Proxy secret with OIDC credentials.
+3. Create a configuration file to be used for installation.
+4. Add an email address to the Keycloak verrazzano-pkce client.
+
+### Create the Keycloak TLS secret
+The existing Keycloak TLS secret needs to be copied from the keycloak namespace to verrazzano-system.  This is a simple procedure as described below.
+
+First create a file containing the existing secret:
+```
+kubectl get secret keycloak-tls -n keycloak -o yaml > ./keycloak-oauth2-tls
+```
+
+Next edit ./keycloak-oauth2-tls and do the following:
+1. change **namespace: keycloak** to **namespace: verrazzano-system**
+2. delete **creationTimestamp** line
+3. delete **resourceVersion** line
+4. delete **uid** line
+
+Finally, create the new secret
+```
+kubectl apply -f ./keycloak-oauth2-tls
+```
+
+Ensure that the secret has been created:
+```
+kubectl get secret -n verrazzano-system | grep keycloak
+```
+output:
+```
+keycloak-oauth2-tls                                     kubernetes.io/tls    3      41h
+```
+
+### Create the OAuth2 Proxy credentials secret
+The oauth2-proxy pod refers to a secret which contains the following fields:
+
+1. client-id
+2. client-secret
+3. cookie-secret
+
+#### create the client id
+The client id, already exists in Keycloak so this will be used.  You can get the clear text client id as follows:
+```
+helm --kubeconfig paul-kubeconfig get values -n verrazzano-system verrazzano-authproxy | grep OIDCClientID
+```
+output:
+```
+  OIDCClientID: <client-id>
+```
+
+Next, base64 encode the client id.  Replace the <...> section with real value from the previous command:
+```
+echo -n <client-id> | base64
+```
+output:
+```
+<client-id-base64>
+```
+
+#### create the client secret
+The client secret is required by the OAuth2 Proxy even though it is not used in this case.  Generate a fake secret using any string, for example:
+
+```
+echo -n fake-secret | base64
```

Comment:
It may be better if you have this step generate a random secret, so that someone following these steps does not copy and paste what you provided.

Reply:
the secret is not even used but i can do that

Reply:
fixed